php/libraries/SinglePointLogin.class.inc

```diff
@@ -328,6 +331,26 @@ class SinglePointLogin
                     return false;
                 }
 
+		// check if the account is longer active
+		$date = new DateTime();
+		$currentDay = $date->getTimestamp();
+		
+		if(($row['active_to'] != NULL) && ($currentDay > strtotime($row['active_to']))){
+			$this->_lastError = "Your account has expired."
+                        . " Please contact your project administrator to re-activate"
+                        . " this account.";
+                	return false;
+
+		}
+
+		// check if the account is active		
+		if(($row['active_from'] != NULL) && ($currentDay < strtotime($row['active_from']))){
+			$this->_lastError = "Your account is not active yet. According to our records it will be active from ".$row['active_from']."."
```

Comment:
You're already using double-quotes here so I think string interpolation is a bit cleaner than concatenating.

```php
$this->_lastError = "Your account is not active yet. According to our records it will be active from {$row['active_from']}."
```